app/app/components/modals/Wallet/WalletList.tsx

```diff
@@ -1,47 +1,47 @@
+import { useState } from "react";
 import Wallet from "./Wallet";
+import { WalletType } from "@/types/global";
+import { SUPPORTED_WALLETS } from "@/utils/wallet/constants";
 
 interface WalletListProps {
-  detected: WalletOption[];
-  undetected: WalletOption[];
-  isWalletValidForChain: (name: string) => boolean;
-  onWalletSelect: (wallet: WalletOption) => void;
+  detected: WalletType[];
+  undetected: WalletType[];
+  isWalletValidForChain: (wallet: WalletType) => boolean;
+  onWalletSelect: (wallet: WalletType) => void;
 }
 
 interface WalletSectionProps {
   title: string;
-  wallets: WalletOption[];
-  isWalletValidForChain: (name: string) => boolean;
-  onWalletSelect: (wallet: WalletOption) => void;
+  wallets: WalletType[];
+  isWalletValidForChain: (wallet: WalletType) => boolean;
+  onWalletSelect: (wallet: WalletType) => void;
 }
 
 const WalletList = ({
-  detected,
-  undetected,
   isWalletValidForChain,
   onWalletSelect,
 }: WalletListProps) => (
   <div className="flex flex-col gap-4">
     <h3 className="text-base text-neutral-900 font-medium font-gt-america">
       Select Wallet
     </h3>
+    <WalletSection
+      title="Detected"
+      wallets={Object.values(SUPPORTED_WALLETS)
+        .filter((wallet) => wallet.isAvailable)
+        .sort((a, b) => a.id.localeCompare(b.id))}
+      isWalletValidForChain={isWalletValidForChain}
+      onWalletSelect={onWalletSelect}
+    />
 
-    {detected.length > 0 && (
-      <WalletSection
-        title="Detected"
-        wallets={detected.sort((a, b) => a.name.localeCompare(b.name))}
-        isWalletValidForChain={isWalletValidForChain}
-        onWalletSelect={onWalletSelect}
-      />
-    )}
-
-    {undetected.length > 0 && (
-      <WalletSection
-        title="Other"
-        wallets={undetected.sort((a, b) => a.name.localeCompare(b.name))}
-        isWalletValidForChain={isWalletValidForChain}
-        onWalletSelect={onWalletSelect}
-      />
-    )}
+    <WalletSection
+      title="Other"
+      wallets={Object.values(SUPPORTED_WALLETS)
```

Comment:
Probably can more efficiently build these lists without doubling the amount of loops

Reply:
Fixed